AWS/s3_access_policy/provider.tf

```diff
@@ -0,0 +1,12 @@
+terraform {
+  experiments = [module_variable_optional_attrs]
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
+    }
+  }
+}
+provider "aws" {
```

Comment:
I think this provider block shouldn't be included here. In the use case, it is likely that we will explicitly set the provider to be used for each call of this module.  
[Terraform documentation](https://www.terraform.io/language/modules/develop/providers) says: 
> A module intended to be called by one or more other modules must not contain any provider blocks.